allennlp/tests/commands/evaluate_test.py

```diff
@@ -1,11 +1,36 @@
 # pylint: disable=invalid-name,no-self-use
 import argparse
 import json
+import torch
+from typing import Iterator, List, Dict
 
 from flaky import flaky
 
-from allennlp.commands.evaluate import evaluate_from_args, Evaluate
+from allennlp.commands.evaluate import evaluate_from_args, Evaluate, evaluate
 from allennlp.common.testing import AllenNlpTestCase
+from allennlp.data import DataIterator
+from allennlp.data.iterators.data_iterator import TensorDict
+from allennlp.models import Model
+
+
+class DummyIterator(DataIterator):
+    def __init__(self, count: int):
+        super().__init__()
+        self._count = count
+
+    def __call__(self, *args, **kwargs) -> Iterator[TensorDict]:
+        while self._count > 0:
+            self._count -= 1
+            yield {}
+
+
+class DummyModel(Model):
+    def __init__(self, outputs: List):
+        super().__init__(None) # type: ignore
+        self._outputs = outputs
+
+    def forward(self, *args, **kwargs) -> Dict[str, torch.Tensor]:  # pylint: disable=arguments-differ
+        return self._outputs.pop(0)
```

Comment:
I think this would be cleaner if `DummyModel` just passed through its inputs and `DummyIterator` supplied the outputs, i.e.

```
class DummyIterator(DataIterator):
    def __init__(self, tensor_dicts: List[TensorDict]) -> None:
        super().__init__()
        self._tensor_dicts = tensor_dicts

     def __call__(self, *args, **kwargs) -> Iterator[TensorDict]:
         yield from self._tensor_dicts

class DummyModel(Model):
    def __init__(self):
        super().__init__() 

     def forward(self, tensor_dict: TensorDict) -> Dict[str, torch.Tensor]:  # pylint: disable=arguments-differ
        return tensor_dict
```

does that still work?

Reply:
Sounds reasonable. I had to modify it slightly given that we unpack the args passed to our models. Ended up with the following:

```
class DummyModel(Model):
    def __init__(self):
        super().__init__(None) # type: ignore

    def forward(self, **kwargs) -> Dict[str, torch.Tensor]:  # pylint: disable=arguments-differ
        return kwargs
```